src/openstudio_lib/RunTabView.cpp

```diff
@@ -257,30 +313,28 @@ void RunView::playButtonClicked(bool t_checked) {
     m_state = State::stopped;
     m_textInfo->clear();
 
-    if (haveConnection) {
-      m_progressBar->setMinimum(0);
-      m_progressBar->setMaximum(State::complete);
-      m_progressBar->setValue(0);
-    } else {
-      m_progressBar->setMinimum(0);
-      m_progressBar->setMaximum(0);
```

Comment:
Setting the maximum to 0 here was to use this feature of the progress bar "If minimum and maximum both are set to 0, the bar shows a busy indicator instead of a percentage of steps. "  This was because I wasn't able to update `m_state` without the socket.  It doesn't look like you are updating `m_state` in the stdout reader so I would suggest leaving this as is.

Reply:
Oh, I see that you are doing that, nevermind this is a good improvement